### PR TITLE
Dogukan/cnx 642 detect selections in Tekla

### PR DESCRIPTION
**Pull request by dogukankaratas**
Adds selection binding and selection filter to the Tekla connector

### CODE REVIEW

File: Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  public string Name => "selectionBinding";
+  public IBrowserBridge Parent { get; }
+
+  public TeklaSelectionBinding(IAppIdleManager idleManager, IBrowserBridge parent)
+  {
+    _idleManager = idleManager;
+    Parent = parent;
+    _events = new Tekla.Structures.Model.Events();
+    _model = new Model();
```
Comment (oguzhankoral) [inline Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs:24]: we have this model initialized in the `SpeckleTeklaPanelHost`, if we will have only one model we can inject it as singleton over there and pass it here via constructor arguments.

File: Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  private readonly Tekla.Structures.Model.UI.ModelObjectSelector _selector;
+
+  public string Name => "selectionBinding";
+  public IBrowserBridge Parent { get; }
+
+  public TeklaSelectionBinding(IAppIdleManager idleManager, IBrowserBridge parent)
+  {
+    _idleManager = idleManager;
+    Parent = parent;
+    _events = new Tekla.Structures.Model.Events();
```
Comment (oguzhankoral) [inline Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs:23]: we could inject it to follow the patterns

File: Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    _model = new Model();
+    _selector = new Tekla.Structures.Model.UI.ModelObjectSelector();
```
Comment (oguzhankoral) [inline Connectors/Tekla/Speckle.Connector.Tekla2024/Bindings/TeklaSelectionBinding.cs:25]: we could inject it to follow the patterns

File: Connectors/Tekla/Speckle.Connector.Tekla2024/HostApp/TeklaIdleManager.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+public sealed class TeklaIdleManager : AppIdleManager
+{
+  private readonly IIdleCallManager _idleCallManager;
+  private readonly Events _events;
+
+  public TeklaIdleManager(IIdleCallManager idleCallManager) : base(idleCallManager)
+  {
+    _idleCallManager = idleCallManager;
+    _events = new Events();
```
Comment (oguzhankoral) [inline Connectors/Tekla/Speckle.Connector.Tekla2024/HostApp/TeklaIdleManager.cs:14]: I saw this also in SendBinding. Can we inject it from the very beginning and pass here via constructor arguments?

File: Connectors/Tekla/Speckle.Connector.Tekla2024/Plugin/TeklaPlugin.cs [hunk 5]
```diff
@@ -8,6 +13,31 @@ namespace Speckle.Connector.Tekla2024.Plugin;
...
+  static TeklaPlugin()
+  {
+    InitializeSpeckle();
+  }
+
+  private static void InitializeSpeckle()
+  {
+    var services = new ServiceCollection();
+
+    services.AddSingleton<ISelectionBinding, TeklaSelectionBinding>();
```
Comment (oguzhankoral) [inline Connectors/Tekla/Speckle.Connector.Tekla2024/Plugin/TeklaPlugin.cs:29]: I would move registrations to `ServiceRegistration` class

**Review by oguzhankoral — CHANGES_REQUESTED** [review]

**Review by clairekuang — CHANGES_REQUESTED** [review]
Need to run a clean-locks and deep-clean to fix package locks

File: Directory.Build.props [hunk 6]
```diff
@@ -60,4 +60,7 @@ complexity threshold -->
     <EmbedUntrackedSources>true</EmbedUntrackedSources>
     <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
   </PropertyGroup>
+  <PropertyGroup>
```
Comment (oguzhankoral) [inline Directory.Build.props:63]: We should remove this here and configure it on Tekla project file

**Review by clairekuang — APPROVED** [review]

**Review by oguzhankoral — APPROVED** [review]

**Pull request merged**